testbed/tests/trace_test.go

```diff
@@ -89,49 +99,86 @@ func TestTrace10kSPS(t *testing.T) {
 	}
 }
 
-func TestTraceNoBackend10kSPSJaeger(t *testing.T) {
-	tests := []struct {
-		name                string
-		configFileName      string
-		expectedMaxRAM      uint32
-		expectedMinFinalRAM uint32
-	}{
-		{name: "NoMemoryLimit", configFileName: "agent-config.yaml", expectedMaxRAM: 200, expectedMinFinalRAM: 100},
-
-		// Memory limiter in memory-limiter.yaml is configured to allow max 10MiB of heap size.
-		// However, heap is not the only memory user, so the total limit we set for this
-		// test is 60MiB. Note: to ensure this test verifies memorylimiter correctly
-		// expectedMaxRAM of this test case must be lower than expectedMinFinalRAM of the
-		// previous test case (which runs without memorylimiter).
-		{name: "MemoryLimiter", configFileName: "memory-limiter.yaml", expectedMaxRAM: 60, expectedMinFinalRAM: 10},
-	}
-
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
+func TestTraceNoBackend10kSPS(t *testing.T) {
 
-			configFilePath := path.Join("testdata", test.configFileName)
+	limitProcessors := map[string]string{
+		"memory_limiter": `
+  memory_limiter:
+   check_interval: 1s
+   limit_mib: 10
+`,
+	}
 
-			tc := testbed.NewTestCase(
-				t,
-				testbed.NewJaegerGRPCDataSender(testbed.DefaultJaegerPort),
-				testbed.NewOCDataReceiver(testbed.DefaultOCPort),
-				testbed.WithConfigFile(configFilePath),
-			)
-			defer tc.Stop()
+	noLimitProcessors := map[string]string{}
 
-			tc.SetResourceLimits(testbed.ResourceSpec{
+	var testSenders = []struct {
+		name          string
+		sender        testbed.DataSender
+		receiver      testbed.DataReceiver
+		resourceSpec  testbed.ResourceSpec
+		configuration []testbed.ProcConfiguration
+	}{
+		{
+			"JaegerThrift",
+			testbed.NewJaegerGRPCDataSender(testbed.DefaultJaegerPort),
+			testbed.NewOCDataReceiver(testbed.DefaultOCPort),
+			testbed.ResourceSpec{
 				ExpectedMaxCPU: 60,
 				ExpectedMaxRAM: 198,
-			})
-
-			tc.StartAgent()
-			tc.StartLoad(testbed.LoadOptions{DataItemsPerSecond: 10000})
-
-			tc.Sleep(tc.Duration)
+			},
+			[]testbed.ProcConfiguration{
+				{
+					Name:                "NoMemoryLimit",
+					Processor:           noLimitProcessors,
+					ExpectedMaxRAM:      200,
+					ExpectedMinFinalRAM: 100,
+				},
+				{
+					Name:                "MemoryLimit",
+					Processor:           limitProcessors,
+					ExpectedMaxRAM:      60,
+					ExpectedMinFinalRAM: 10,
+				},
+			},
+		},
```

Comment:
This seems to be the same for Jaeger and Zipkin, so would be nice to extract as a var and reuse.